### PR TITLE
185 nbhood dimensionality

### PR DESCRIPTION
**Pull request by bayliffe**
#185 Restore dimensionality of cubes after neighbourhooding. Added unit tests for utility and updated CLI acceptance test data.

- [x] Tests pass
- [x] New tests added

### CODE REVIEW

File: lib/improver/nbhood/nbhood.py [hunk 1]
```diff
@@ -147,6 +147,55 @@ def adjust_nsize_for_ens(ens_factor, num_ens, width):
...
+        iris.exceptions.CoordinateNotFoundError raised if the final dimension
+        coordinates of the returned cube do not match the input cube.
+
+        """
+        # Promote available and relevant scalar coordinates
+        for coord in new_cube.aux_coords[::-1]:
+            if coord in cube.dim_coords:
+                new_cube = iris.util.new_axis(new_cube, coord)
+
+        # Ensure dimension order matches; if lengths are unequal a coordinate
```
Comment (fionaRust) [inline lib/improver/nbhood/nbhood.py]: Looking at the acceptance tests input and output you are losing the bounds on your dimension coordinate through the processing chain. I suggest we have a look. I've put an ipython notebook link on JIRA.

File: lib/improver/nbhood/nbhood.py [hunk 2]
```diff
@@ -147,6 +147,55 @@ def adjust_nsize_for_ens(ens_factor, num_ens, width):
...
+        ----------
+        cube : iris.cube.Cube
+            The input cube provided to nbhood.
+        new_cube : iris.cube.Cube
+            The cube produced by the neighbourhooding process that must be
+            checked for demoted dimensional coordinates.
+
+        Returns
+        -------
+        new_cubewidth : iris.cube.Cube
```
Comment (bayliffe) [inline lib/improver/nbhood/nbhood.py]: :white_check_mark:

File: lib/improver/nbhood/nbhood.py (hunk 1 above)
Comment (bayliffe) [inline lib/improver/nbhood/nbhood.py]: This was the case before this change. We can fix it here, but it is really a different technical debt issue.

Comment (bayliffe) [inline lib/improver/nbhood/nbhood.py]: Fixed.

**Comment by fionaRust** [comment]
All tests pass on desktop and new unit tests have been added.
Plotted the acceptance tests and they look fine and have the right dimensions when appropriate.

**Pull request merged**